std/range/package.d

```diff
@@ -6911,21 +6911,27 @@ Indexed!(Source, Indices) indexed(Source, Indices)(Source source, Indices indice
 
 /**
 This range iterates over fixed-sized chunks of size $(D chunkSize) of a
-$(D source) range. $(D Source) must be a forward range. $(D chunkSize) must be
+$(D source) range. $(D Source) must be an input range. $(D chunkSize) must be
 greater than zero.
 
 If $(D !isInfinite!Source) and $(D source.walkLength) is not evenly
 divisible by $(D chunkSize), the back element of this range will contain
 fewer than $(D chunkSize) elements.
 
+If `Source` is an input range but not a forward range, the resulting range and
+chunks will be single-pass only: iterating over `front` will shrink the chunk
```

Comment:
Well, I just wanted to make sure people come with no expectations that we will cache range elements or otherwise make the resulting range "better behaved".  A *lot* of Phobos algorithms will act funny when given a non-forward input range (try it sometime), and I have heard complaints in the past about why something doesn't behave "intuitively", even though that is impossible given input range semantics.